go.mod

```diff
@@ -7,5 +7,6 @@ require (
 	github.com/mitchellh/cli v1.0.0
 	github.com/mitchellh/go-homedir v1.1.0
 	github.com/sirupsen/logrus v1.4.2
+	github.com/stretchr/testify v1.2.2
```

Comment:
I would like to see us stick to the standard library as much as possible.

Reply:
@theckman , something more like this? Where I use `reflect` for the comparison in an if statement. 

```
func TestAbility_ListAbilities(t *testing.T) {
	t.Parallel()
	setup()
	defer teardown()

	mux.HandleFunc("/abilities", func(w http.ResponseWriter, r *http.Request) {
		testMethod(t, r, "GET")
		w.Write([]byte(`{"abilities": ["sso"]}`))
	})

	var client = &Client{apiEndpoint: server.URL, authToken: "foo", HTTPClient: defaultHTTPClient}
	want := &ListAbilityResponse{Abilities: []string{"sso"}}

	res, err := client.ListAbilities()

	if err != nil {
		t.Fatal(err)
	}

	if !reflect.DeepEqual(res, want) {
		t.Errorf("returned %#v; want %#v", res, want)
	}
}
```

Also, as for the size of the PR, adding testing is no small task. But didn't seem I should do in pieces. In hindsight, might have been smarter to propose a testing style and then go forward.